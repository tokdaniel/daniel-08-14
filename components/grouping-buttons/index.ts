export { default as GroupingButtons } from './grouping-buttons'
