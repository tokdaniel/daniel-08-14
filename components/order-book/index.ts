export { default as OrderBook } from './order-book'
export { default as HalfBook } from './half-book'
