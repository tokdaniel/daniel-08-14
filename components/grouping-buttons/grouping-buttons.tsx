import React from 'react'
import styled from 'styled-components'

import { Text } from 'components/typography'

const Container = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row-reverse;
`

const Button = styled.button`
  font-size: 20px;
  background: transparent;
  border: none;
  cursor: pointer;
  border-radius: 4px;
  color: ${({ theme }) => theme.palette.grey[100]};

  :hover {
    background: rgba(33, 33, 33, 0.6);
  }
`

type Props = {
  selected: number
  increment: () => void
  decrement: () => void
  groupings?: number[]
}

const GroupingButtons: React.FC<Props> = ({
  selected,
  increment,
  decrement,
}) => {
  return (
    <Container>
      <Text variant="label">group: {selected}</Text>
      <Button onClick={increment}>+</Button>
      <Button onClick={decrement}>âˆ’</Button>
    </Container>
  )
}

export default GroupingButtons
