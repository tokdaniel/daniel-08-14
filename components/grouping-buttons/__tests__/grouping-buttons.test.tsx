import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

import { GroupingButtons } from 'components/grouping-buttons'
import theme from 'config/theme.config'

describe('GroupingButtons', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <GroupingButtons
            increment={() => null}
            decrement={() => null}
            selected={25}
          />
          ,
        </ThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
