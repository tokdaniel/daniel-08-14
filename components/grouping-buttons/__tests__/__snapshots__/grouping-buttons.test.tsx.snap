// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupingButtons renders correctly 1`] = `
Array [
  <div
    className="sc-gtsrHT dVvDsX"
  >
    <div
      className="sc-bdnxRM jDkucz"
    >
      group: 
      25
    </div>
    <button
      className="sc-dlnjwi gTMfzH"
      onClick={[Function]}
    >
      +
    </button>
    <button
      className="sc-dlnjwi gTMfzH"
      onClick={[Function]}
    >
      âˆ’
    </button>
  </div>,
  ",",
]
`;
